toolkit/scripts/sodiff/mariner-sodiff.sh

```diff
@@ -9,41 +9,93 @@ repo_file_path="$2"
 mariner_version="$3"
 sodiff_out_dir="$4"
 sodiff_log_file="${sodiff_out_dir}/sodiff.log"
+current_os=$(cat /etc/os-release | grep ^ID | cut -d'=' -f2)
+temporary_repo_file="file.repo"
 
 # Setup output dir
 mkdir -p "$sodiff_out_dir"
+# Empty the log file
+echo > "$sodiff_log_file"
 
-# Prepare mariner/ubuntu compatibility calls
+function makecache_with_common {
+    if [[ "$current_os" == mariner ]]; then
+        # Mariner uses DNF repoquery command
+        DNF_COMMAND=dnf
+        # Cache RPM metadata
+        >/dev/null dnf $common_options -y makecache
+    else
+        # Ubuntu uses repoquery command from yum-utils
+        DNF_COMMAND=
+        # Cache RPM metadata
+        # Ubuntu does not come with gpgcheck plugin for yum
+        >/dev/null yum $common_options -y --nogpgcheck makecache
+    fi
+}
+
+function log_to_file {
+    >>$sodiff_log_file echo -- $@
+}
+
+# Split the abrdiged repo file and process each repo separately
+# because if any of the repos is not accessible,
+# creating cache for the whole .repo file fails.
+# However, created cache combines , so creating multiple caches
+# Will not overwrite each other and any failures can be safely ignored
```

Comment:
In what scenarios do we want to ignore such failures? I'm curious if there are some repos we expect to be always able to cache and should error-out in case of a failure.

Reply:
Our current case is that a repo is available for only one of two architectures.